src/KafkaFlow/ConsumerManagerFactory.cs

```diff
@@ -22,7 +22,7 @@ public IConsumerManager Create(IConsumerConfiguration configuration, IDependency
                 configuration,
                 logHandler);
 
-            consumerWorkerPool.WorkerPoolStopped.Subscribe(middlewareExecutor);
+            consumerWorkerPool.WorkerPoolStopped.Subscribe(() => middlewareExecutor.OnNotification());
```

Comment:
rename the method `OnNotification` to **OnWorkerPoolStopped()** please, and change it to be a private method

Reply:
Done